qiime2/sdk/proxy.py

```diff
@@ -67,42 +88,46 @@ def _get_element_(self, results):
         """
         return getattr(results, self._selector_)
 
+    def export_data(self, output_dir):
+        return self.result().export_data(output_dir)
+
     def save(self, filepath, ext=None):
         """Blocks then calls save on the result.
         """
         return self.result().save(filepath, ext=ext)
 
+    def validate(self, level=NotImplemented):
+        return self.result().validate(level=level)
+
 
 class ProxyArtifact(ProxyResult):
     """This represents a future Artifact that is being returned by a Parsl app
     """
-    def __repr__(self):
-        if self._signature_ is None:
-            return f'<artifact: Unknown Type {object.__repr__(self)}>'
-        else:
-            return \
-                f'<artifact: {self._signature_[self._selector_].qiime_type}>'
-
     def view(self, type):
         """If we want to view the result we need the future to be resolved
         """
         return self._get_element_(self._future_.result()).view(type)
 
+    def has_metadata(self):
```

Comment:
You could replace this with `has_metadata = Artifact.has_metadata` which will get bound to this class in the usual way.

Reply:
Shenanigans

Reply:
Unfortunately, getting some circular imports here trying to implement these shenanigans